src/pymor/algorithms/tr.py

```diff
@@ -0,0 +1,464 @@
+                elif current_output - estimate_output > compare_output:
+                    # reject new mu
+                    rejected = True
+                    # shrink the radius
+                    radius *= shrink_factor
+
+                    msg = 'Estimated output larger than previous output.'
+                else:
+                    surrogate.extend(mu)
+                    current_output = surrogate.new_output(mu)
+                    if current_output <= compare_output:
+                        # fom.output is potentially for free after enrichment, e.g. using caching
+                        current_fom_output = fom.output(mu)
+                        fom_output_diff = old_fom_output - current_fom_output
+                        rom_output_diff = old_rom_output - current_output
+                        if fom_output_diff >= radius_tol * rom_output_diff:
+                            # increase the radius if the model confidence is high enough
+                            radius /= shrink_factor
+
+                        msg = 'Updated model output smaller than previous output.'
+                    else:
+                        # reject new mu
+                        rejected = True
+                        # shrink the radius
+                        radius *= shrink_factor
+
+                        msg = 'Updated model output larger than previous output.'
+
+            # handle parameter rejection
+            if not rejected:
+                data['mus'].append(mu.copy())
+                mu_norm = np.linalg.norm(mu)
+                update_norms.append(np.linalg.norm(mu - old_mu))
+
+                data['subproblem_data'].append(sub_data)
+
+                with logger.block('Computing first order criticality...'):
+                    # fom.output_d_mu is potentially for free after enrichment, e.g. using caching
+                    gradient = fom.parameters.parse(fom.output_d_mu(mu)).to_numpy()
+                    first_order_criticality = np.linalg.norm(mu - parameter_space.clip(mu - gradient).to_numpy())
+                    foc_norms.append(first_order_criticality)
+
+                surrogate.accept()
+                logger.info(f'Current mu iterate accepted: {msg}')
+
+                old_rom_output = current_output
+            else:
+                mu = old_mu
+                surrogate.reject()
+                logger.info(f'Current mu iterate rejected: {msg}')
+
+            if not np.isfinite(mu_norm):
+                raise TRError('Failed to converge.')
+
+    logger.info('')
+
+    data['update_norms'] = np.array(update_norms)
+    data['foc_norms'] = np.array(foc_norms)
+    data['iterations'] = iteration
+    data['rom_evaluations'] = surrogate.rom_evaluations
+    data['enrichments'] = surrogate.enrichments
+
+    return mu, data
+
+def CoerciveRB_trust_region(fom, surrogate, parameter_space=None, initial_guess=None, beta=.95, radius=.1,
```

Comment:
```suggestion
def coercive_rb_trust_region(fom, surrogate, parameter_space=None, initial_guess=None, beta=.95, radius=.1,
```

The other should be in violation of PEP8